static/app/views/projectDetail/charts/projectErrorsBasicChart.tsx

```diff
@@ -1,134 +1,109 @@
-  location: Location;
+type Props = {
   onTotalValuesChange: (value: number | null) => void;
-  organization: Organization;
   projectId?: string;
 };
 
-type State = DeprecatedAsyncComponent['state'] & {
-  projects: Project[] | null;
-};
-
-class ProjectErrorsBasicChart extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      projects: null,
-    };
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, projectId} = this.props;
-
-    if (!projectId) {
-      return [];
-    }
-
-    return [
-      [
-        'projects',
-        `/organizations/${organization.slug}/projects/`,
-        {
-          query: {
-            statsPeriod: this.getStatsPeriod(),
-            query: `id:${projectId}`,
-          },
-        },
-      ],
-    ];
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    const {location} = this.props;
-    if (!ERRORS_BASIC_CHART_PERIODS.includes(location.query.statsPeriod)) {
-      browserHistory.replace({
-        pathname: location.pathname,
+function ProjectErrorsBasicChart({projectId, onTotalValuesChange}: Props) {
+  const organization = useOrganization();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const urlStatsPeriod = searchParams.get('statsPeriod') ?? '';
+  const statsPeriod = ERRORS_BASIC_CHART_PERIODS.includes(urlStatsPeriod)
+    ? urlStatsPeriod
+    : DEFAULT_STATS_PERIOD;
+
+  const {
+    data: projects,
+    isLoading,
+    isError,
+    isSuccess,
+  } = useApiQuery<Project[]>(
+    [
+      `/organizations/${organization.slug}/projects/`,
+      {
         query: {
-          ...location.query,
-          statsPeriod: this.getStatsPeriod(),
-          start: undefined,
-          end: undefined,
+          statsPeriod,
+          query: `id:${projectId}`,
         },
+      },
+    ],
+    {
+      staleTime: 0,
+      enabled: defined(projectId),
+    }
+  );
+  const stats = useMemo(() => {
+    return projects?.[0]?.stats ?? [];
+  }, [projects]);
+  const totalValues = stats.reduce((acc, [, value]) => acc + value, 0);
+
+  useEffect(() => {
+    if (!ERRORS_BASIC_CHART_PERIODS.includes(urlStatsPeriod)) {
+      setSearchParams(oldParams => {
+        oldParams.set('statsPeriod', DEFAULT_STATS_PERIOD);
+        oldParams.delete('start');
+        oldParams.delete('end');
+        return oldParams;
       });
     }
-  }
-
-  onLoadAllEndpointsSuccess() {
-    this.props.onTotalValuesChange(
-      this.state.projects?.[0]?.stats?.reduce((acc, [, value]) => acc + value, 0) ?? null
-    );
-  }
-
-  getStatsPeriod() {
-    const {location} = this.props;
-    const statsPeriod = location.query.statsPeriod;
+  }, [setSearchParams, urlStatsPeriod]);
 
-    if (ERRORS_BASIC_CHART_PERIODS.includes(statsPeriod)) {
-      return statsPeriod;
+  useEffect(() => {
+    if (isSuccess) {
+      onTotalValuesChange(totalValues);
```

Comment:
yes the classic pass the result back up. makes sense

Reply:
no idea what it's doing with this info but I do hate it